atintegrators/attrackfunc.c

```diff
@@ -15,11 +15,17 @@ void trackRFCavity(double *r_in, double le, double nv, double freq, double h, do
 {
     int c;
 
+    /* If nv is 0 and length is 0, then skip this whole loop (good for passive rf cavities
+        anyway if there is a cavity length, we have to loop through the particles
+    */
+
     if (le == 0) {
-        for (c = 0; c<num_particles; c++) {
-            double *r6 = r_in+c*6;
-            if(!atIsNaN(r6[0]))
-                r6[4] += -nv*sin(TWOPI*freq*((r6[5]-lag)/C0 - (h/freq-T0)*nturn) - philag);
+        if (nv != 0) {
+            for (c = 0; c<num_particles; c++) {
+                double *r6 = r_in+c*6;
+                if(!atIsNaN(r6[0]))
+                    r6[4] += -nv*sin(TWOPI*freq*((r6[5]-lag)/C0 - (h/freq-T0)*nturn) - philag);
+            }
         }
     }
     else {
```

Comment:
Line 39
`if(nv!=0.0) r6[4] += -nv*sin(TWOPI*freq*((r6[5]-lag)/C0 - (h/freq-T0)*nturn) - philag);`

Would save you a sine calculation

Reply:
done